### ISSUE TITLE
`SubmitProvenTransaction` does not error on already-spent notes

### ISSUE THREAD

**Issue by igamigo** [0xPolygonMiden/miden-node#249]
Currently, a client can double-spend a note by executing 2 consecutive similar transactions and not syncing in between (if there was a sync and the client knows the note was consumed, we can simply error before attempting execution). What happens on the node is the following: `SubmitProvenTransaction` does not error out (ie, it does not check double-spending at the RPC call), but the block builder will then fail iteratively, bringing the node into a failure state since it won't be able to construct blocks anymore. 

The error is the following:

````
2024-02-26T14:33:58.016077Z  INFO try_build_block:build_block:execute_program: miden_processor: .cargo/git/checkouts/miden-vm-3db6be6805d1979c/c718a0a/processor/src/lib.rs:119: close, time.busy: 30.3ms, time.idle: 54.1µs
2024-02-26T14:28:48.008614Z ERROR try_build_block:build_block: miden-block-producer: block-producer/src/block_builder/mod.rs:69: error: failed to compute new block: program execution failed
````

**Comment by polydez** [comment]
@igamigo @bobbinth @hackaugusto @plafer  this bug is much wider: any transaction with incorrect proof will stay in the queue forever and stop the whole block building process. This is actually a vulnerability, when any malformed/malicious transaction may stop the block producer from producing new blocks.

**Why it happens:**

If block producer fails to produce block due to error (i.e. proof checking failed), it puts the whole bunch of current batches back to the queue. It doesn't distinguish, what kind of error occurred and how to deal with it and it even doesn't know, which transaction was malformed. Thus, any transaction which can't be processed, will stop processing of the whole block and whole block building will be repeated more and more.

I think, we should:
1. Categorize error kinds on batches/blocks building as recoverable or not. I guess, the most of errors are unrecoverable.
2. For unrecoverable errors, we should find malformed transaction and exclude it from the block. It's not a straightforward task, how to mitigate damage from such transaction. I think, we can use something similar to binary searching, i.e. bisect and reassemble batches and try to prove their parts.

Any ideas?

**Comment by polydez** [comment]
And, of course, we should make more preflights checks in order to discard such transaction(s) ASAP.

**Comment by hackaugusto** [comment]
duplicate: https://github.com/0xPolygonMiden/miden-node/issues/235

**Comment by hackaugusto** [comment]
It seems there is a bug in the state view. The [conflicting transactions](https://github.com/0xPolygonMiden/miden-node/blob/ba25321abcbe8c545f022d67f1704436743ed94f/block-producer/src/state_view/mod.rs#L173-L181) should not have been added to the TxQueue.

**Comment by bobbinth** [comment]
agreed with @hackaugusto - the wider issue would be addressed as a part of fixing #235. For this issue we need to fix the specific bug where a transaction which tries to consume an already consumed note should not get into the transaction queue in the first place.
